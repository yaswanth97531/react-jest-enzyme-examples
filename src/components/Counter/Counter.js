import React, { Component } from 'react';

class Counter extends Component {
  state = {
    count : 0
  }

  increment = () => {
    this.setState(prevState => {
      return {
        count: prevState.count + 1
      }
    });
  }

  decrement = () => {
    this.setState(prevState => {
      return {
        count: prevState.count - 1
      }
    });
  }

  render() {
    return (
      <div>
        <p>Current count: { this.state.count }</p>
        <button id='increment' onClick={this.increment}>+++</button>
        <button id='decrement' onClick={this.decrement}>---</button>
      </div>
    );
  }
}

export default Counter;