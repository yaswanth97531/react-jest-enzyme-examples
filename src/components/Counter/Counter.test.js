import React from 'react'
import Counter from './Counter';
import { exportAllDeclaration } from '@babel/types';

let wrapper;

beforeEach(() => {
  wrapper = shallow( <Counter / > );
});

describe('<Counter /> rendering', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('starts with a count of 0', () => {
    const text = wrapper.find('p').text();
    expect(text).toEqual('Current count: 0');
  });
});

describe('<Counter /> interactions', () => {
  it('can increment the count when the button is clicked', () => {
    const incrementBtn = wrapper.find('#increment')
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current count: 1')
  });
  
  it('can decrement the count when the button is clicked', () => {
    const decrementBtn = wrapper.find('#decrement')
    decrementBtn.simulate('click');
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current count: -1')
  });
});

describe('<Counter /> lifecycle method invocations', () => {});