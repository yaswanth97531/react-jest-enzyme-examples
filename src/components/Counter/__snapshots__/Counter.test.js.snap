// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> rendering matches the snapshot 1`] = `
<div>
  <p>
    Current count: 
    0
  </p>
  <button
    id="increment"
    onClick={[Function]}
  >
    +++
  </button>
  <button
    id="decrement"
    onClick={[Function]}
  >
    ---
  </button>
</div>
`;
